### PR TITLE
[OD-2083] Merge upstream master

### PR DESCRIPTION
**Pull request by jguo144**
## Description
Merge upstream master

Also makes the following changes:
- Remove use of second tracking id (googleanalytics.id2)
- Removes Package and Organization from Downloads events
- Import local validator_args function on ImportError (<= CKAN 2.7)
- Fix lint issues

### CODE REVIEW

**Review by surajdeval — APPROVED** [review]
LGTM

File: ckanext/googleanalytics/controller.py [hunk 1]
```diff
@@ -146,45 +127,28 @@ def search(self, ver=None, register=None):
...
-                "dh": c.environ['HTTP_HOST'],
-                "dp": c.environ['PATH_INFO'],
-                "dr": c.environ.get('HTTP_REFERER', ''),
-                "ec": "CKAN Organization Page View",
-                "ea": request_obj_type + request_function,
-                "el": request_id,
-            }
-            plugin.GoogleAnalyticsPlugin.analytics_queue.put(data_dict)
-
-        if toolkit.config.get('googleanalytics.id2'):
```
Comment (peterVorman) [inline ckanext/googleanalytics/controller.py:165]: Looks like we removing necessary functionality here.

Comment (jguo144) [reply ckanext/googleanalytics/controller.py:165]: We were not actively using the second Google Analytics tracking id so we're removing it.

**Review by peterVorman — APPROVED** [review]
LGTM

**Pull request merged**